processing/src/main/java/org/apache/druid/math/expr/Function.java

```diff
@@ -4318,6 +4362,24 @@ public ExprEval<?> apply(List<Expr> args, Expr.ObjectBinding bindings)
           return ExprEval.ofLongBoolean(false);
         }
       }
+
+      @Nullable
+      @Override
+      public BitmapColumnIndex asBitmapColumnIndex(ColumnIndexSelector selector, List<Expr> args)
```

Comment:
What testing strategy do you have in mind for stuff like this? I suppose we could add items to `ExpressionFilterTest`. But ideally it would be possible to test them in `FunctionTest`. What do you think about extending that to have an `assertBooleanExpr` that calls `asBitmapColumnIndex` with an index selector that has indexes for the bindings?

With that, I'd also suggest updating `assertExpr` to check if the expr has a boolean return type, and throw if it does (so they all get moved to `assertBooleanExpr`).